contracts/modules/Liquidation.sol

```diff
@@ -99,61 +119,142 @@ contract Liquidation is BaseLogic {
 
             liqOpp.baseDiscount = baseDiscount;
             liqOpp.discount = discount;
+
             liqOpp.conversionRate = liqLocs.underlyingPrice * 1e18 / liqLocs.collateralPrice * 1e18 / (1e18 - discount);
         }
 
         // Determine amount to repay to bring user to target health
 
-        if (liqLocs.underlying == liqLocs.collateral) {
-            liqOpp.repay = type(uint).max;
-        } else {
-            AssetConfig memory collateralConfig = resolveAssetConfig(liqLocs.collateral);
-            AssetConfig memory underlyingConfig = resolveAssetConfig(liqLocs.underlying);
+        OverrideConfig memory overrideConfig;
+        AssetConfig memory collateralConfig;
+        AssetConfig memory underlyingConfig;
+
+        collateralConfig = resolveAssetConfig(liqLocs.collateral);
+        underlyingConfig = resolveAssetConfig(liqLocs.underlying);
+
+        uint collateralFactor = collateralConfig.collateralFactor;
+        uint borrowFactor = underlyingConfig.borrowFactor;
+
+        // If override is active for the liquidated pair, assume the resulting liability will be fully covered by override collateral, and adjust inputs
+        if (liqLocs.overrideCollateralValue > 0) {
+            overrideConfig = overrideLookup[liqLocs.underlying][liqLocs.collateral];
+            // the liquidated collateral has active override with liability
+            if (overrideConfig.enabled || liqLocs.underlying == liqLocs.collateral) {
+                collateralFactor = overrideConfig.enabled ? overrideConfig.collateralFactor : selfCollateralFactor;
+                borrowFactor = CONFIG_FACTOR_SCALE;
+                // adjust the whole liability for override BF = 1
+                liqLocs.liabilityValue = liqLocs.currentOwed * liqLocs.underlyingPrice / 1e18;
+            }
+        }
+
+        // Calculate for no overrides or resulting liability fully covered by override
+        // or if liquidating non-override collateral assume result will be partially covered by override
+        calculateRepayCommon(liqOpp, liqLocs, collateralFactor, borrowFactor);
```

Comment:
what happens in a scenario where:
- there's no override
- there was a sudden drop of collateral value or sudden increase of liability value
- in `calculateRepayCommon`, due to the sudden price change, the following is `true`: `liabilityValue <= liqLocs.collateralValue`?

it seems that the `liqOpp.repay` after `calculateRepayCommon` will be 0 and the end of  `computeLiqOpp` will be reached with that value, while intuitively it should be maxed out, shouldn't it?

Reply:
In case there's no overrides, then the account composition is 3.1. and note that `liabilityValue - liqLocs.collateralValue` is the numerator in that equation, which is guaranteed to be positive, as explained in the doc, so this condition won't trigger. It might trigger when the BF is swapped for 1 in 3.3., in which case the secondary calc should be carried out. This is ensured by setting repay to 0 and testing for it in `(liqOpp.repay == 0 || // numerator in equation was negative`

Reply:
verified this and agree